### PR TITLE
RD-3460 Select only one environment on GettingStartedModal

### PR DESCRIPTION
**Pull request by NorbertAmartus**
## Description

Modifications are allowing the user to select only one environment from the environment list visible in `GettingStartedModal`.

While creating changes I was not focused on the refactors related to the code (that's why there are eg. still methods which relate on the array of selected environments), because I had a discussion with the UI Lead about how the business may wants to eventually bring back this functionality and how this modal may be recreated in a not distant future.
So having that in mind, we agreed that refactors related to adjusting the "guts" of the modal should be avoided.

## Screenshots / Videos

N/A

## Checklist

- [x] My code follows the UI Code Style.
- [x] I verified that all tests and checks have passed.
- [x] I verified if that change requires a documentation update.
- [x] I added proper labels to this PR.

## Tests

 

Link to the [System tests](https://jenkins.cloudify.co/view/UI%20Health%20View/job/Stage-UI-System-Test/1610)
Additionally, I've successfully executed the `test:frontend` tests

## Documentation

N/A

### CODE REVIEW

**Comment by NorbertAmartus** [comment]
NOTE:
One of the test scenarios (of the `getting_started_spec.ts` file) looks unstable
I've created a [ticket](https://cloudifysource.atlassian.net/browse/RD-3692) which provides more details about the mentioned issue.
I tried to fix it, but it was taking too much time, that's why I thought about spliting this issue to a separate ticket

**Comment by NorbertAmartus** [comment]
NOTE:
All files of the `GettingStartedModal` are fully converted to the `typescript`, and all of them are connected to te translation file.
That's why in the PR we can't see any changes related to the `typescript` / `translation` refactors

File: test/cypress/integration/getting_started_spec.ts [hunk 1]
```diff
@@ -332,6 +306,27 @@ describe('Getting started modal', () => {
         });
     });
 
+    it('only enables one environment to be selected at a time', () => {
```
Comment (qooban) [inline test/cypress/integration/getting_started_spec.ts:309]: Maybe:
```suggestion
    it('should allow only one environment to be selected at a time', () => {
```
would be more accurate in this context ("allow" sounds for me better suited than "enable") as well as be consistent with other tests naming?

File: test/cypress/integration/getting_started_spec.ts [hunk 2]
```diff
@@ -332,6 +306,27 @@ describe('Getting started modal', () => {
...
+        goToNextStep();
+        cy.contains('button', 'AWS').click();
+        cy.contains('button.active', 'AWS');
+
+        cy.contains('button', 'GCP').click();
+        cy.contains('button.active', 'AWS').should('not.exist');
+        cy.contains('button.active', 'GCP');
+    });
+
+    it('enables to click "next" button when an environment is being selected', () => {
```
Comment (qooban) [inline test/cypress/integration/getting_started_spec.ts:319]: Maybe to be consistent with other test cases naming it'd be more accurate to have:
```suggestion
    it('should allow to click "Next" button when an environment is being selected', () => {
```
? ("allow" sounds for me better suited than "enable")

File: test/cypress/integration/getting_started_spec.ts [hunk 3]
```diff
@@ -332,6 +306,27 @@ describe('Getting started modal', () => {
...
+        cy.contains('button', 'Next').should('to.be.disabled');
+
+        cy.contains('button', 'AWS').click();
+        cy.contains('button', 'Next').should('not.to.be.disabled');
+
+        cy.contains('button', 'AWS').click();
+        cy.contains('button', 'Next').should('to.be.disabled');
+    });
+
     it('should keep button and field states when navigating beetwen steps', () => {
```
Comment (qooban) [inline test/cypress/integration/getting_started_spec.ts:330]: It's not your change, but maybe worth to fix at once:
```suggestion
    it('should keep button and field states when navigating between steps', () => {
```

File: test/cypress/integration/getting_started_spec.ts [hunk 4]
```diff
@@ -233,61 +233,40 @@ describe('Getting started modal', () => {
...
+            cy.wait(toAliasReferences(awsBlueprints));
 
-            verifyInstallationSucceeded(blueprints);
+            verifyInstallationSucceeded(awsBlueprints);
             closeModal();
         });
     });
 
-    it('requires all secrets to go to next step', () => {
+    it('requires secret to go to the summary step', () => {
```
Comment (qooban) [inline test/cypress/integration/getting_started_spec.ts:269]: Even if we have just one technology selected, we still can have multiple secrets to provide 🙂, so I think having plural ("all secrets") here is still OK.

**Review by qooban — COMMENTED** [review]
Tested UI - works fine 👍🏻 

I have only very minor comments related to tests naming convention.

File: test/cypress/integration/getting_started_spec.ts (hunk 1 above)
Comment (NorbertAmartus) [reply test/cypress/integration/getting_started_spec.ts:309]: I agree, that looks more suitable :)
I'll provide that inside the upcoming commit

File: test/cypress/integration/getting_started_spec.ts (hunk 2 above)
Comment (NorbertAmartus) [reply test/cypress/integration/getting_started_spec.ts:319]: Seems good to me,
I'll provide that inside the upcoming commit

File: test/cypress/integration/getting_started_spec.ts (hunk 3 above)
Comment (NorbertAmartus) [reply test/cypress/integration/getting_started_spec.ts:330]: Sure, that change wouldn't take the whole sprint, so why not 😁

File: test/cypress/integration/getting_started_spec.ts (hunk 4 above)
Comment (NorbertAmartus) [reply test/cypress/integration/getting_started_spec.ts:269]: Hmm, we can think about it in that way, so sure, I'll provide those changes :)

File: test/cypress/integration/getting_started_spec.ts (hunk 1 above)
Comment (NorbertAmartus) [reply test/cypress/integration/getting_started_spec.ts]: Done

File: test/cypress/integration/getting_started_spec.ts (hunk 2 above)
Comment (NorbertAmartus) [reply test/cypress/integration/getting_started_spec.ts]: Done

File: test/cypress/integration/getting_started_spec.ts (hunk 3 above)
Comment (NorbertAmartus) [reply test/cypress/integration/getting_started_spec.ts]: Done

File: test/cypress/integration/getting_started_spec.ts (hunk 4 above)
Comment (NorbertAmartus) [reply test/cypress/integration/getting_started_spec.ts]: Done

**Review by qooban — APPROVED** [review]
LGTM 👍🏻

**Pull request merged**